rspace/src/main/scala/coop/rchain/rspace/spaces/FineGrainedReplayRSpace.scala

```diff
@@ -371,10 +371,14 @@ class FineGrainedReplayRSpace[F[_], C, P, E, A, R, K](store: IStore[C, P, A, K],
     }
   }
 
-  override def clear(): F[Unit] = syncF.delay {
-    replayData.clear()
-    super.clear()
-  }
+  override def clear(): F[Unit] =
+    syncF
+      .delay {
+        replayData.clear()
+      }
+      .flatMap { _ =>
+        super.clear()
+      }
```

Comment:
maybe a `for` would make this more readable?

Reply:
Yeah +1 on that `for/yield`